Reactivating moving the dock

dr-hc-solo opened the pull request:
Dock can be moved around again. It will stick to all sides (lrtb) of the window. It saves it's position in local store. Also more than four participants take less space now.

oskarkrawczyk commented:
@dr-hc-solo Thanks for this one! There's some chance to deps in this PR, I'll have to verify all deps resolve correctly before pulling in

FANMixco commented:
Where is mobile.less? I cannot see it.

Commit:
check all .emit calls

Commit:
remove unmatched closing comment

Commit:
calls to updateFromHello typecheck

Commit:
put checks in TODO.md

Commit:
.

Commit:
fix some type errors in many places

Commit:
fix most errors in peers.ts

Commit:
fix most problems in forms.ts

Commit:
replace type with a shorter and direct identifier

Commit:
almost fix ot.ts

Commit:
some typing for ui.ts, still a lot to do

Commit:
fix all errors in walkthrough.ts

Commit:
can't really type webrtc.ts more, remaining errors can be fixed only by changing the code

Commit:
remove walkabout.ts

Commit:
add esprima.js

Commit:
add YT lib

Commit:
finished typing youtubeVideo.ts

Commit:
update todo

Commit:
fix errors into channels.ts

Commit:
recording prefix for storage.get

Commit:
add declarations for walkabout

Commit:
fix url of types dir in tsconfig

Commit:
fix some types for storage

Commit:
conditional type for Storage

Commit:
hide some unused errors

Commit:
update todo

Commit:
add comment marking a line that can be removed

Commit:
remove ref to mixinEvents in the core
the function has been moved into testutils.js, the only place where it's still used

Commit:
fix jqueryPlugins.ts

Commit:
some new and mostly deprecated types to remove errors from webrtc
a lot of explanations were added in the comments

Commit:
no more errors in webrtc.ts

Commit:
fix simple errors in some files

Commit:
windowing.ts without errors

Commit:
no error in who.ts

Commit:
no more errors in video.ts

Commit:
change UpdatePeer type to everything optional which seems to match the usecase

Commit:
playback.ts has no error
The playback feature has been tested and works

Commit:
remove errors from chat (probably introduced by previous changes)

Commit:
clearer code for logMessage and fix some types

Commit:
fix message in recorder.ts

Commit:
add some fields to some message and type some more

Commit:
towards fixing forms.ts and the string|boolean problem of getValue

Commit:
type for tinymce

Commit:
throw when tiny editor is undefined

Commit:
string or boolean in forms.ts solved
MessageForEditor now carry information about the presence and type of basis, tracker and element with the introduction of many subtypes that transcribe the constraints

Commit:
write explanations about MessageForEditor and put everything into a namespace for easier use and clearer code

Commit:
remove errors in templating.ts

Commit:
fix some errors in startup.ts

Commit:
remove some unused variables, no more error in recorder.ts

Commit:
fix once and callback errors with a lot of casts

Commit:
add some TODO comments about how to make bette .on(ce) and removed an unused function

Commit:
fix lots of errors in tjs.ts

Commit:
fix syntax error

Commit:
only 1 error in tjs.ts

Commit:
fix some errors in ui.ts

Commit:
fix some errors in ui.ts

Commit:
remove alert

Commit:
add comment about hidden stuff in interface

Commit:
almost no error in ui.ts
all tests passing

Commit:
fix error in jqueryPlugin

Commit:
fix some easy errors

Commit:
.

Commit:
no more errors in peers.ts

Commit:
correct types for peer in many places but it leads to a lot of errors

Commit:
remove ExternalPeer from PeerView

Commit:
explain the whole thing about PeerView.update() being called with a this.peer of type PeerSelf

Basically it can never happen to we just return when isSelf is true

Commit:
before BasePeerView as a PeerView for PeerSelf

Commit:
introduce PeerSelfView, no error in ui.ts, all tests pass!

Commit:
only 1 error left in session.ts

Commit:
comment 2 unused var in session.ts

Commit:
no more error in session.ts

Commit:
move OnMap into On.Map

Commit:
prepare before splitting on calls

Commit:
calls to on splitted

Commit:
split .on() calls with multi events, type check ok

Commit:
a bad typing in templating.sub was hiding an error, fixed

Commit:
prefix unused var with _

Commit:
fix error in templating.sub

Commit:
fix some unused var errors

Commit:
PeerSelfView can now be create from an ExternalPeer, some hacks were used to do that (like having fields of type only undefined in external peer) and pass typechecking but it should be ok

Commit:
fix ts-lint comment

Commit:
fix the problem of Change/string in ot, everything was actually string

Commit:
No more errors!
hide the last error with @ts-expect-error, it shouldn't matter for now and it's require related which I want to remove anyway

Commit:
clean togetherjs.d.ts and split the last .on("a b c") I forgot

Commit:
cleaning some TODOs

Commit:
only create SimpleHistory if the value of the form is of type string, also getValue now returns "" when the actual value is null

Commit:
Went through all TODO comments and resolve the really easy ones, some code has been removed in consequence but nothing major

Commit:
.

Commit:
remove invalid TODO comment

Commit:
update TODO.md

Commit:
reformat all "} else" and "} catch"

Commit:
update TODO

Commit:
reformat TODO.md with checkboxes

Commit:
fix typos in TODO.md

Commit:
trying to make jquery not global

Commit:
.

Commit:
remove log

Commit:
hack to set JQuery as global in the tests

Commit:
order of arguments were wrong in eventmakerMain

Commit:
rename eventmakerMain into eventMakerMain

Commit:
update todo with local jquery

Commit:
migrate to tinycolor2

Commit:
tinycolor2 working

Commit:
tinycolor removed

Commit:
update TODO

Commit:
almost all imports working except with Randomizer

Commit:
add random.js

Commit:
no import error but does not load

Commit:
try to make togetherjs.ts not a module, without success so far

Commit:
add mimetypes to devserver

Commit:
use import type

Commit:
works better
go to rawtjs.html and do require(["../main"]); TogetherJS.start();

Commit:
add new files

Commit:
.

Commit:
try to rewrite in devserver

Commit:
remove some function in togetherjs.ts

Commit:
set some fields as public

Commit:
somewhere (over the rainbow)

Commit:
add a working version of rawtjs

Commit:
fix type error

Commit:
go back to previous togetherjs.ts

Commit:
global declarations with imports

Commit:
missing import jquery

Commit:
fix require calls in tests, we need to do require("session").session now

Commit:
bind assert

Commit:
fix some export with a new syntax and remove main.ts file

Commit:
rawtjs works

Commit:
trying to fix jquery problem

Commit:
import statements for jquery

Commit:
misc tests passing with modification

Commit:
all tests passing!

Commit:
Merge branch 'jsImports2' into develop

Commit:
fix rawtjs.html

Commit:
.

Commit:
add link to jquery

Commit:
update gitignore

Commit:
mark import system as done

Commit:
move togetherjs build output to the build/togetherjs directory

Commit:
remove alert

Commit:
working tsc compilation

Commit:
comment unused var

Commit:
fix grunt file

Commit:
add notes about browserify

Commit:
reformat gruntfile

Commit:
remove alerts

Commit:
add jquery dep to tests

Commit:
clean tests, can't manege to maje automatic tests with phantomjs work :/

Commit:
cannot reproduce bug

Commit:
delete empty "client" file

Commit:
Merge branch 'develop' into bugManydata

Commit:
tried something

Commit:
some cleanup

Commit:
reduce the number of casts

it appers that using callbackForOn by default instead of cbForOnce makes more sense

Commit:
improved OnClass for custom protocol

Commit:
working protocol (I think)

Commit:
working and correctly typed PacketAssembler

cannot reproduce bug yet

Commit:
add js file

Commit:
remove ugly exports from the code base

Commit:
better exports and imports

Commit:
remove alias to JQuery

Commit:
util use global jq

Commit:
remove Module class

Commit:
getContext("2d")!, ! is ok because the first call to getContext can't fail if it's "2d"

From https://developer.mozilla.org/en-US/docs/Web/API/HTMLCanvasElement/getContext

"Later calls to this method on the same canvas element, with the same contextType argument, will always return the same drawing context instance as was returned the first time the method was invoked. It is not possible to get a different drawing context object on a given canvas element."
And "2d" context is always implemented, contrary to "webgl" for example.

Commit:
add comment for getContext

Commit:
lazy loading of feedback form

Commit:
do not load youtube api by default

Commit:
improve hub

Commit:
typings for require to avoid mistakes

Commit:
explain overload in Require itnerface

Commit:
fix Require overloads for walkabout

Commit:
remove a todo for getContext("2d")!

Commit:
configure eslint

Commit:
try to fix some rules

Commit:
fix some eslint errors

Commit:
fix some eslint errors

Commit:
solve most eslint problems

Commit:
go back on some eslint fixes for now and fix tests

"ot.ot." was no longer necessary in tests

Commit:
all easily fixable lint fixed

Commit:
some eslint config

Commit:
remove clone and replace with Object.assign

Commit:
no need to polyfill console

Commit:
grunt

Commit:
fix interactive test

Commit:
fix interactive tests again

Commit:
remove weird check for RTCSupported that was only relevant for firefox 18 and 19

Commit:
no need to collapse the dock, it will be done in CSS at one point

Commit:
.

Commit:
remove weird sharing feature that was enabeling bluetooth

Commit:
replace any by void in callback type

Commit:
trying to clean console.ts

Commit:
since console.ts has been removed the related tests have been disabled

Commit:
remove ref to console module from tjs.d.ts

Commit:
disable some lints in backward-compat.d.ts

Commit:
void instead of any in some callbacks and remove type that weren't used

Commit:
move global TogetherJS decl at the top of the file

Commit:
remove unused channels and related types in channels.ts

Commit:
disable no-var rule for TogetherJS declaration

Commit:
remove references to TowTruck

Commit:
move TogetherJS decl around

Commit:
move _configClosed into configClass

Commit:
add some parameters to TJS's constructor

Commit:
simpler configuration logic

remove _defaultCOnfiguration, close function, ...

Commit:
pass startup to TJS constructor

Commit:
move hubBase higher in the code

Commit:
.

Commit:
remove _loaded

Commit:
requireObject improved

Commit:
reformat comment and remove commented code

Commit:
move _configTrackers into ConfigClass

Commit:
group fields by visibility

Commit:
remove startTarget since it wasn't used anywhere

Commit:
remove code related to startTarget

Commit:
rename _startupInit to startupInit

Commit:
move configuration into ConfigClass

had to introduce a has function on ConfigFunObj to check if a string was a valid configuration key

Commit:
set pageLoaded ad readonly

Commit:
format code

Commit:
change how copyConfig works

it does not copy into the window anymore but into the actualConfiguration variable

Commit:
clean logic for configOverride

Commit:
reorder functions

Commit:
move ConfigClass outside of togetherjsMain

Commit:
move createConfigFunObj to top level

Commit:
move TogetherJSClass to top level

Commit:
remove getConfig and replace with config.get

Commit:
.

Commit:
start refactoring Router

Commit:
remove Route.send since it wasn't used

Commit:
remove some comments

Commit:
add public keyword

Commit:
format

Commit:
.

Commit:
remove ConfigFunObj

Commit:
remove configFunObj from tjs.d.ts

Commit:
make peers.ts a little more pure

Commit:
purer cursor

Commit:
reformat some comments and move readyForMessages into Session class

Commit:
clean ui.ts

Commit:
format comment

Commit:
format comment

Commit:
put "TODO !!" where removing the non null assertion is a priority

Commit:
format comment

Commit:
templates function now returns a HTMLElement

Commit:
fix templates and callback for startup (wrong this context)

Commit:
more strict visiblity for fields in OnClass

Commit:
fix code where the leading "app." would be removed by mistake

Commit:
starting to fix some type errors with ts 4.5.4

Commit:
finish fix compat issues with ts 4.5.5

Commit:
add some comments

Commit:
Reactivated moving of the dock

Commit:
Merge branch 'move_dock' of https://github.com/dr-hc-solo/togetherjs into move_dock

# Conflicts:
#	togetherjs/jqueryPlugins.js
#	togetherjs/storage.js
#	togetherjs/togetherjs.less
#	togetherjs/ui.js
#	togetherjs/windowing.js

Commit:
rollback old js files
